### ISSUE TITLE
`atom --test` not showing exceptions, simply exiting with no message

### ISSUE THREAD

**Issue by lee-dohm** [atom/atom#15868]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

When running `atom --test spec` and there is a problem encountered that will not allow the tests to start, this is the result:

```
[master][~/Source/atom/keybinding-resolver] atom --test spec
Error Downloading Update: Could not get code signature for running application
[master][~/Source/atom/keybinding-resolver][1]
```

It turned out that the problem was that I had forgotten to run `apm install` on the repository before attempting to run the tests. When I finally attempted to run the tests in the GUI, I was able to find this in the JavaScript Console

```
Error: Cannot find module 'etch'
  at Module._resolveFilename (module.js:470:15)
  at Module._resolveFilename (/Applications/Atom.app/Contents/Resources/electron.asar/common/reset-search-paths.js:35:12)
  ...
```

This led me to the solution of running `apm install`. We should output any exception like this at the command line to help people figure out why things aren't working.

### Steps to Reproduce

1. Execute `git clone https://github.com/atom/keybinding-resolver.git ~/keybinding-resolver-test`
1. `cd ~/keybinding-resolver-test`
1. `atom --test spec`

**Expected behavior:** An exception message (or even better a real error message explaining the problem)

**Actual behavior:** Program simply exits with an unhelpful error code

### Versions

Atom    : 1.23.0-dev-dec47761f
Electron: 1.6.14
Chrome  : 56.0.2924.87
Node    : 7.4.0

apm  1.18.8
npm  3.10.10
node 6.9.5 x64
python 2.7.10
git 2.14.2

**Comment by thomasjo** [comment]
Confirming that I experience the same problem on Ubuntu 17.04 (as well as macOS 10.12).
Worth noting that the behavior is (naturally) the same for `apm test`.

My immediate guess is that something has changed/regressed in Electron, and that we inherited this via a version bump. It could also be something else — trying to investigate further now.

**Comment by thomasjo** [comment]
This regressed due to https://github.com/atom/atom/pull/15148. The temporary workaround is to use the flag `--enable-electron-logging`. The proper fix involves setting the environment variable `ENABLE_ELECTRON_LOGGING` when the user is running tests. Exactly where that line of code needs to be introduced is unknown to me at this point in time, but likely not in `atom.sh` since this also needs to apply to `apm test`.

**Comment by Arcanemagus** [comment]
Came here to file this bug only to find it was already filed for me 😉.

For anyone coming to this, if you run the specs from within an Atom window the developer tools should pop open with the error that would normally be printed but is silenced due to this bug.

**Comment by skylize** [comment]
The supposed "proper fix" suggested by @thomasjo will not be sufficient, because currently when `enable-electron-logging` is enabled, _the test results are not shown at all_, and the log output is couched inside a lot of useless noise.

**Comment by Arcanemagus** [comment]
Closing this as it _should_ be fixed by #18863, we can reopen later if this turns out to be wrong.
